scalafix-core/shared/src/main/scala/scalafix/internal/patch/EscapeHatch.scala

```diff
@@ -357,28 +332,6 @@ object EscapeHatch {
         rulesToPos.result()
       }
 
-      def trackUnusedRules(
-          anchor: Token.Comment,
-          rules: String,
-          enabled: Boolean): Unit = {
-        val rulesToPos = rulesWithPosition(splitRules(rules), anchor)
-
-        if (enabled) {
-          if (currentlyDisabledRules.isEmpty && rulesToPos.isEmpty) { // wildcard
-            unusedAnchoredEnable += anchor.pos
-          } else {
-            val disabledRules = currentlyDisabledRules.keySet
-            rulesToPos.foreach {
-              case (rule, pos) if !disabledRules(rule) =>
-                unusedAnchoredEnable += pos
-              case _ => ()
-            }
-          }
-        } else {
-          currentlyDisabledRules ++= rulesToPos
-        }
-      }
-
```

Comment:
the main change is here. `trackUnusedRules` is being replaced by `OnOffTracker` (see below).

In the old code once a rule is added to `currentlyDisabledRules`, it is never removed from it. This prevents potential unused `scalafix:on` comments further down the file from being reported as unused.